project-base/storefront/components/Pages/Order/ContactInformation/ContactInformationDeliveryAddress.tsx

```diff
@@ -72,71 +65,49 @@ export const ContactInformationDeliveryAddress: FC = () => {
-                setValue(formMeta.fields.deliveryTelephone.name, deliveryAddress?.telephone ?? '');
-                if (pickupPlace === null) {
-                    setValue(formMeta.fields.deliveryStreet.name, deliveryAddress?.street ?? '');
-                    setValue(formMeta.fields.deliveryCity.name, deliveryAddress?.city ?? '');
-                    setValue(formMeta.fields.deliveryPostcode.name, deliveryAddress?.postcode ?? '');
-                    setValue(
-                        formMeta.fields.deliveryCountry.name,
-                        selectedCountryOption ?? countriesAsSelectOptions[0],
-                    );
+            if (countriesAsSelectOptions.length) {
+                // setValue(formMeta.fields.deliveryAddressUuid.name, deliveryAddressUuidValue);
+                setValue(formMeta.fields.deliveryFirstName.name, deliveryAddress.firstName);
+                setValue(formMeta.fields.deliveryLastName.name, deliveryAddress.lastName);
+                setValue(formMeta.fields.deliveryCompanyName.name, deliveryAddress.companyName);
+                setValue(formMeta.fields.deliveryTelephone.name, deliveryAddress.telephone);
+                setValue(formMeta.fields.deliveryCountry.name, selectedCountryOption);
+
+                if (!pickupPlace) {
+                    setValue(formMeta.fields.deliveryStreet.name, deliveryAddress.street);
+                    setValue(formMeta.fields.deliveryCity.name, deliveryAddress.city);
+                    setValue(formMeta.fields.deliveryPostcode.name, deliveryAddress.postcode);
                 }
             }
+        } else if (deliveryAddressUuid === '') {
+            setValue(formMeta.fields.deliveryFirstName.name, '');
+            setValue(formMeta.fields.deliveryLastName.name, '');
+            setValue(formMeta.fields.deliveryCompanyName.name, '');
+            setValue(formMeta.fields.deliveryTelephone.name, '');
+            setValue(formMeta.fields.deliveryStreet.name, '');
+            setValue(formMeta.fields.deliveryCity.name, '');
+            setValue(formMeta.fields.deliveryPostcode.name, '');
+            setValue(formMeta.fields.deliveryCountry.name, countriesAsSelectOptions[0]);
         }
-    }, [
-        countriesAsSelectOptions,
-        deliveryAddressUuidValue,
-        formMeta.fields.deliveryCity.name,
-        formMeta.fields.deliveryCompanyName.name,
-        formMeta.fields.deliveryCountry.name,
-        formMeta.fields.deliveryFirstName.name,
-        formMeta.fields.deliveryLastName.name,
-        formMeta.fields.deliveryPostcode.name,
-        formMeta.fields.deliveryStreet.name,
-        formMeta.fields.deliveryTelephone.name,
-        user,
-        pickupPlace,
-        setValue,
-        user?.country.code,
-        user?.deliveryAddresses,
-    ]);
+    }, [deliveryAddressUuid, countriesAsSelectOptions]);
 
-    if (countriesAsSelectOptions.length === 0) {
+    if (!countriesAsSelectOptions.length) {
         return null;
     }
```

Comment:
I know we talked about this before, but this change seems like its for worse. Why was it necessary? You are only handling one case, that is the length being zero. Why not handle it explicitly?